src/ideascale/fetch.rs

```diff
@@ -0,0 +1,113 @@
+use crate::ideascale::models::de::{Fund, Funnel, Proposal};
+
+use once_cell::sync::Lazy;
+use serde::de::DeserializeOwned;
+use serde::Deserialize;
+use url::Url;
+
+use std::collections::HashMap;
+use std::convert::TryInto;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    RequestError(#[from] reqwest::Error),
+
+    #[error("Could not get value from json, missing attribute {attribute_name}")]
+    MissingAttribute { attribute_name: &'static str },
+}
+
+#[derive(Debug, Deserialize)]
+struct Score {
+    #[serde(alias = "ideaId")]
+    id: i32,
```

Comment:
Not using `u32` means the idea ID could be negative?

Reply:
No it can not, it was just most convenient that way since database already work with signed intergers so not to cast them one another.

Reply:
Actually, those ids are not used finally except for internal linking of data, so I can fairly use unsigned ones.